app/src/main/java/org/mozilla/focus/screenshot/ScreenshotViewerActivity.java

```diff
@@ -453,13 +421,18 @@ public boolean onLongClick(View v) {
 
     private void proceedDelete() {
         if (mScreenshot != null) {
-            File file = new File(mScreenshot.getImageUri());
-            if (file.exists()) {
-                try {
-                    file.delete();
-                } catch (Exception ex) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    File file = new File(mScreenshot.getImageUri());
+                    if (file.exists()) {
+                        try {
+                            file.delete();
+                        } catch (Exception ex) {
+                        }
+                    }
                 }
-            }
+            }).start();
             ScreenshotManager.getInstance().delete(mScreenshot.getId(), this);
```

Comment:
How about deleting the file in the callback?

Reply:
It seems fine to delete the file in `onDeleteComplete()`. However, I prefer to keep it here in this commit to focus on fixing the strict mode issue. What's the benefit you think to delete the file in the callback?